File: uniffi_bindgen/src/bindings/dart/templates/Int64Helper.dart
class _UniffiConverterInt64 extends _UniffiConverterPrimitiveInt  {

    int read(_UniffiRustBufferBuilder buf) {
        return buf.read_i32();
    }

    _UniffiRustBufferBuilder write(int value) {
      Pointer<_UniffiRustBuffer> pointer = calloc<_UniffiRustBuffer>();
      pointer.ref
        ..capacity = 8
        ..len = 8
        ..data = "0000".toNativeUtf8();

        return pointer.ref.buffer;
    }
}

File: uniffi_bindgen/src/bindings/dart/templates/Int64Helper.py
class _UniffiConverterInt64(_UniffiConverterPrimitiveInt) {
    CLASS_NAME = "i64"
    VALUE_MIN = -2**63
    VALUE_MAX = 2**63

    static read(buf) {
        return buf.read_i64()
    }

    static write_unchecked(value, buf) {
        buf.write_i64(value)
    }
}
